frontend/src/hooks/useDropdownResizeObserver.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect, Dispatch, SetStateAction } from 'react';
+import ResizeObserver from 'resize-observer-polyfill';
+
+export type UseDropdownResizeObserverProps = {
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+  dropDownRefElement: HTMLElement | null;
+};
+
+/**
+ *  Custom hook that observes the window height changes and closes the dropdown
+ *  if the height falls below a certain threshold.
+ *  @param {UseDropdownResizeObserverProps} props - The props for the hook.
+ *  @property {Dispatch<SetStateAction<boolean>>} setIsOpen - Sets the state for the component.
+ *  @property {HTMLElement | null} dropDownRefElement - The HTML element used to compare the height to the window height.
+ */
+const useDropdownResizeObserver = ({
```

Comment:
Is it possible to make a test for this new hook?

Reply:
I thought about it at first but then that maybe it won't be worth the effort. given that you brought it up, too, I'll work on adding some

Reply:
+1 I think it's worth the effort, shouldn't be too hard either IMO